bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/workbench/renderers/swt/StackRenderer.java

```diff
@@ -623,6 +637,66 @@ public Object createWidget(MUIElement element, Object parent) {
 		return tabFolder;
 	}
 
+	/**
+	 * @param tabFolder
+	 */
+	private void readExtensionPointAndSetText(OnboardingCTabFolder tabFolder) {
+		Composite container = WidgetFactory.composite(SWT.None).layout(GridLayoutFactory.swtDefaults().create())
+				.create(tabFolder);
+
+		GridDataFactory gridData = GridDataFactory
+				.create(GridData.VERTICAL_ALIGN_CENTER | GridData.HORIZONTAL_ALIGN_CENTER).grab(true, true);
+
+		Composite content = WidgetFactory.composite(SWT.NONE).supplyLayoutData(gridData::create)
+				.layout(GridLayoutFactory.swtDefaults().create()).create(container);
+
+		Color color = container.getDisplay().getSystemColor(SWT.COLOR_WIDGET_DISABLED_FOREGROUND);
+		Label welcomeText = WidgetFactory.label(SWT.NONE).foreground(color).supplyLayoutData(gridData::create)
+				.create(content);
+
+		Label welcomeImage = WidgetFactory.label(SWT.NONE).foreground(color).supplyLayoutData(gridData::create)
+				.create(content);
+
+		tabFolder.setOnboarding(container);
+
+		IPageService service = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getService(IPageService.class);
```

Comment:
I think it is easier to use the IEventBroker here and you don't require the org.eclispe.ui dependency in this  case

Reply:
done, but I don't get informed with the EventBroker. Maybe subscription is too late. But nevertheless I guess I should get informed when the perspective is changed?!